### PR TITLE
Search packs one by one in the build install packs step

### PR DESCRIPTION
**Pull request by daryakoval**
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-6327
Change the way we search packs in the build. Instead of searching each pack a-synchrony, let's search them and their dependencies one by one.

### CODE REVIEW

File: Tests/Marketplace/search_and_install_packs.py [hunk 1]
```diff
@@ -690,18 +690,27 @@ def search_and_install_packs_and_their_dependencies(pack_ids: list,
 
     lock = Lock()
 
-    with ThreadPoolExecutor(max_workers=130) as pool:
+    if not install_packs_one_by_one:
```
Comment (tkatzir) [inline Tests/Marketplace/search_and_install_packs.py:693]: Why not change it to be more straight forward, and positive? :-)
`if not install_packs_one_by_one` -> `if install_packs_one_by_one`
And swap the `if` and `else` blocks?

**Review by tkatzir — APPROVED** [review]
Approved, with a comment.

File: Tests/Marketplace/search_and_install_packs.py [hunk 2]
```diff
@@ -690,18 +690,27 @@ def search_and_install_packs_and_their_dependencies(pack_ids: list,
...
+        with ThreadPoolExecutor(max_workers=130) as pool:
+            for pack_id in pack_ids:
+                if is_pack_hidden(pack_id):
+                    logging.debug(f'pack {pack_id} is hidden, skipping installation and not searching for dependencies')
+                    continue
+                pool.submit(search_pack_and_its_dependencies,
```
Comment (tkatzir) [inline Tests/Marketplace/search_and_install_packs.py:699]: You're not waiting for them to finish?

**Review by tkatzir — APPROVED** [review]
Approved, with comments.

File: Tests/Marketplace/search_and_install_packs.py (hunk 1 above)
Comment (daryakoval) [reply Tests/Marketplace/search_and_install_packs.py:693]: Positive is me! I will change it thanks.

File: Tests/Marketplace/search_and_install_packs.py (hunk 2 above)
Comment (daryakoval) [reply Tests/Marketplace/search_and_install_packs.py:699]: `with ThreadPoolExecutor` context manager handles it. (waiting to finish on exit).

**Pull request merged**